cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/GoalOptimizer.java

```diff
@@ -117,6 +118,37 @@ public GoalOptimizer(KafkaCruiseControlConfig config,
     _proposalComputationTimer = dropwizardMetricRegistry.timer(MetricRegistry.name("GoalOptimizer", "proposal-computation-timer"));
   }
 
+  /**
+   * Generate unique goal priority for each of proposal precomupting thread
+   */
+  private void populateGoalByPriorityForPrecomputing() {
+    Set<List<Goal>> shuffledGoalSet = new HashSet<>();
+    //Need at least one computing thread
+    for (int i = 0; i < numProposalComputingThreads(); i++) {
```

Comment:
Hi Efe, I think you are right. if _numPrecomputingThreads equals 0, we should only generate one goalByPriority.
Actually I am confused by the numProposalComputingThreads(). based on its comments,

 At least two computing thread is needed if precomputing is disabled. One thread for submitting and waiting for the proposal computing to finish, another one for compute the proposals.

when precomputing is disabled. in GoalOptimizer.computeBestProposal().

    for (int i = 0; i < (_numPrecomputingThreads > 0 ? _numPrecomputingThreads : 1); i++) {
      futures.add(_proposalPrecomputingExecutor.submit(new ProposalCandidateComputer(_goalByPriorityForPrecomputing.get(i))));
    }

does this mean that the actual thread for submitting and waiting for the proposal computing to finish is the main thread instead of one of thread managed by       _proposalPrecomputingExecutor?

I tried to change numProposalComputingThreads do
  private int numProposalComputingThreads() {
    return _numPrecomputingThreads > 0 ? _numPrecomputingThreads : 1;
  }
and is still able to pass all the tests.
I am going to have this in the PR. Please correct me if I am wrong.